### PR TITLE
test: enabling Spark tests with offHeap requirement

### PR DESCRIPTION
**Pull request by kazuyukitanimura**
## Rationale for this change

After https://github.com/apache/datafusion-comet/pull/1062 We have not running Spark tests for native execution

## What changes are included in this PR?

Removed the off heap requirement for testing

## How are these changes tested?

Bringing back Spark tests for native execution

### CODE REVIEW

File: spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala [hunk 1]
```diff
@@ -921,8 +921,8 @@ class CometSparkSessionExtensions
     override def apply(plan: SparkPlan): SparkPlan = {
 
       // Comet required off-heap memory to be enabled
-      if ("true" != conf.getConfString("spark.memory.offHeap.enabled", "false")) {
-        logInfo("Comet extension disabled because spark.memory.offHeap.enabled=false")
+      if (!isOffHeapEnabled(conf) && !isTesting) {
+        logWarning("Comet extension disabled because spark.memory.offHeap.enabled=false")
```
Comment (parthchandra) [inline spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala:925]: Should we add this to extended explain info?

Comment (kazuyukitanimura) [reply spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala]: added

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @kazuyukitanimura

**Comment by kazuyukitanimura** [comment]
Merged, thanks @andygrove

**Pull request merged**